language-examples/remedial-js/11.0-event-qeue/README.md

```diff
@@ -0,0 +1,44 @@
+
+# 11.0 - Example Event Qeue
+
+# Purpose:
+
+  - An overview of how the browser event queue works
+  - What it means to post something to the event queue
+  - How user interactions are also processed through the event queue
+
+### Overview
+
+Running applications in a web browser involves a collection of technologies, such as a JavaScript Engine (such as Chrome’s V8), a collection of Web API’s (such as the DOM), and the Event Loop and Event Queue.
+
+### What exactly does the JavaScript Engine do?
+
+It’s job is to go through all the lines of JavaScript in an application and process them one at a time. This means that JavaScript is single-threaded. The main repercussion of this is that if you’re running a line of JavaScript that happens to take a long time to return, then all the code after that will be blocked.
+
+**So how does javascript engine know how to process a single line of javascript at a time?**
+
+It uses a call stack. You can think of a call stack like entering an elevator — the first person who enters the elevator is the last person to exit the elevator, whereas the last person to enter is the first to exit.
+
+### Where does the event qeue and loop play into this?
+
+The way Javascript avoids blocking code is that the engine provides a mechanism and it’s via asynchronous callback functions.
+
+An asynchronous callback function is just like any other function you’re used to writing in JavaScript, with the added caveat that it doesn’t get executed till later.
+
+### How is a function posted to the event qeue?
+
+When an async function is executed (like setTimeout), something special happens to its callback — the browser places the callback function into an Event Table. Think of the event table as a registration booth: the call stack tells the event table to register a particular function to be executed only when a specific event happens.
```

Comment:
the wording of this is strange. Apart from ES7 async functions, I don't think there is a such thing as an async function in javascript, just async APIs (`requestAnimationFrame`, `XMLHttpRequest`, `setTimeout`...). `setTimeout()` executes synchronously and pushes another completely synchronous function onto the event queue.